subprojects/docs/src/samples/userguide/mavenMigration/profiles/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,10 @@
+val buildProfile: String? by project  // <1>
+
+apply(from = "profile-${buildProfile ?: "default"}.gradle.kts")  // <2>
+
+val message: String by extra
```

Comment:
:o: I'd move the extra property val closer to its usage inside the `doLast {}` block below. Note that it would have to be written `val message: String by project.extra` in order not to query the Task for its extra property but the Project.

Reply:
I wanted to, but realised that it would become an extra task property. Thanks for the necessary syntax to overcome that.